### PR TITLE
[MM-15300] Migrate "Preference.PermanentDeleteByUser" to Sync by default

### PR DESCRIPTION
**Pull request by Wipeout55**
#### Summary

This PR migrates PermanentDeleteByUser in the Preference store to use a synchronous call by default.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/10717

### CODE REVIEW

File: store/sqlstore/preference_store.go [hunk 1]
```diff
@@ -216,13 +216,16 @@ func (s SqlPreferenceStore) GetAll(userId string) store.StoreChannel {
...
 
-func (s SqlPreferenceStore) PermanentDeleteByUser(userId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if _, err := s.GetMaster().Exec(
-			`DELETE FROM Preferences WHERE UserId = :UserId`, map[string]interface{}{"UserId": userId}); err != nil {
-			result.Err = model.NewAppError("SqlPreferenceStore.Delete", "store.sql_preference.permanent_delete_by_user.app_error", nil, err.Error(), http.StatusInternalServerError)
-		}
-	})
+func (s SqlPreferenceStore) PermanentDeleteByUser(userId string) *model.AppError {
+	if _, err := s.GetMaster().Exec(
```
Comment (hanzei) [inline store/sqlstore/preference_store.go]: Maybe it increases readability if this is written like:
```go
query := `DELETE FROM 
			Preferences 
		WHERE 
			UserId = :UserId`
if _, err := s.GetMaster().Exec(query); err != nil {
```

Comment (hanzei) [reply store/sqlstore/preference_store.go]: @Wipeout55 Would you please fix this and then we can merge this PR?

Comment (Wipeout55) [reply store/sqlstore/preference_store.go]: Done! Sorry for being slow.

**Comment by jespino** [comment]
Thanks @Wipeout55!, merged into master. :heart:

**Comment by jespino** [comment]
@hanzei I don't see the need of cherry-picking this into 5.12 release.

**Comment by hanzei** [comment]
I agree.

**Pull request merged**